lib/moonshot/stack.rb

```diff
@@ -154,12 +155,14 @@ def default_values
     end
 
     def template
-      @template ||= StackTemplate.new(template_file, log: @log)
+      @template ||= load_template_file
     end
 
     # @return [String] the path to the template file.
     def template_file
-      File.join(Dir.pwd, 'cloud_formation', "#{@app_name}.json")
+      json = "#{raw_template_file_name}.json"
+      yaml = "#{raw_template_file_name}.y*ml"
+      @template_file ||= Dir[json].first || Dir[yaml].first || raise
```

Comment:
We can cut 0.8.0 to signify the breaking change. However, if we're going to do that I wonder if we want to even support this type of access. Instead, we'd probably want to act as an abstraction layer for the plugins, so they don't have to do File I/O and serialize/deserialize. Plus we're caching the file now, meaning if they do change the file later actions won't be affected. This would definitely break any plugins using that kind of modification of the template.

I don't see that we bundle any plugins that use this feature, nor is it used in the one internal plugin I know of. I feel comfortable saying we'd rather provide interfaces to specific data (like `#resources` and `#parameters` in this class), than give plugins the file to muck with directly, anyway.